src/runtime/qurt_thread_pool.cpp

```diff
@@ -117,60 +115,23 @@ WEAK int halide_do_par_for(void *user_context,
         // initializing this mutex.
         qurt_mutex_init(mutex);
     }
-
-    wrapped_closure c = {closure, qurt_hvx_get_mode()};
-
-    // Set the desired number of threads based on the current HVX
-    // mode.
-    int old_num_threads =
-        halide_set_num_threads((c.hvx_mode == QURT_HVX_MODE_128B) ? 2 : 4);
+    int old_num_threads = halide_set_num_threads(HEXAGON_DEFAULT_NUM_THREADS);
     // We're about to acquire the thread-pool lock, so we must drop
     // the hvx context lock, even though we'll likely reacquire it
     // immediately to do some work on this thread.
-    if (c.hvx_mode != -1) {
-        // The docs say that qurt_hvx_get_mode should return -1 when
-        // "not available". However, it appears to actually return 0,
-        // which is the value of QURT_HVX_MODE_64B!  This means that
-        // if we enter a do_par_for with HVX unlocked, we will leave
-        // it with HVX locked in 64B mode, which then never gets
-        // unlocked (a major bug).
-
-        // To avoid this, we need to know if we are actually locked in
-        // 64B mode, or not locked. To do this, we can look at the
-        // return value of qurt_hvx_unlock, which returns an error if
-        // we weren't already locked.
-        if (qurt_hvx_unlock() != QURT_EOK) {
-            c.hvx_mode = -1;
-        }
-    }
-    int ret = halide_default_do_par_for(user_context, task, min, size, (uint8_t *)&c);
+    qurt_hvx_unlock();
```

Comment:
Oh I see, we unlock HVX here instead. This is kinda sketchy to have half of this logic in CodeGen_Hexagon and half of it here.